core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -837,6 +848,32 @@ func (vdb *VersionedDB) GetFullScanIterator(skipNamespace func(string) bool) (st
 	return nil, byte(0), errors.New("Not yet implemented")
 }
 
+// ChannelMetadataExists returns true if channel metadata already exists in VersionedDB
+func (vdb *VersionedDB) ChannelMetadataExists() bool {
+	return vdb.channelMetadata != nil
+}
+
+// CreateChannelMetadata builds channelMetadata and stores in channel metadata db
+func (vdb *VersionedDB) CreateChannelMetadata(namespaces []string) error {
+	vdb.mux.Lock()
+	defer vdb.mux.Unlock()
+	if vdb.channelMetadata != nil {
+		return errors.Errorf("channel metadata already existed for channel %s", vdb.chainName)
+	}
+	dbsInfo := make(map[string]*namespaceDBInfo, len(namespaces))
+	for _, ns := range namespaces {
+		dbsInfo[ns] = &namespaceDBInfo{
+			Namespace: ns,
+			DBName:    constructNamespaceDBName(vdb.chainName, ns),
+		}
+	}
+	vdb.channelMetadata = &channelMetadata{
+		ChannelName:      vdb.chainName,
+		NamespaceDBsInfo: dbsInfo,
+	}
+	return vdb.writeChannelMetadata()
```

Comment:
Shouldn't we check for the database existence and filter certain namespaces?

By default, we create a database for a chaincode or collection only when it is accessed first (applicable for implicit collection too). Hence, if we blindly create mapping for the received namespaces and write the channelMetadata, it would end up with inconsistent details. To avoid this inconsistency, we need to check whether a db exist for a given namespace and only then add it to the channel metadata.